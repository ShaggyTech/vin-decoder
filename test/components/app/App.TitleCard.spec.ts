// import flushPromises from 'flush-promises';

import { mount } from '@vue/test-utils';

import AppTitleCard from '@/components/app/TitleCard.vue';

const factory = (options: { [propName: string]: any } = {}) => {
  return mount(AppTitleCard, {
    ...options,
  });
};

describe('AppTitleCard Component Tests', () => {
  // let wrapper: ReturnType<typeof factory>;

  test('base component state, matches snapshot', () => {
    const wrapper = factory({
      sync: false,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  test('Props "header" and "subheader"', () => {
    const wrapper = factory({
      sync: false,
      propsData: {
        header: {
          innerText: 'Heading',
          class: 'some-class',
        },
        subHeader: {
          innerText: 'Sub Heading',
          class: 'some-other-class',
        },
      },
    });
    // console.log(wrapper.vm);
    // expect(wrapper.vm.$el).toEqual('display-1');
    const header = wrapper.find('#header');
    console.log({ classList: header.element.classList });
    expect(wrapper.find('#header').exists()).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
});
