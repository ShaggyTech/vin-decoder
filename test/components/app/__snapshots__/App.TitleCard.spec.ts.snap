// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitleCard Component Tests Props "header" and "subheader" 1`] = `
<div
  class="v-toolbar__title"
>
  <nuxt-link-stub
    to="/"
  >
    <span
      class="some-class"
      id="header"
      innertext="Heading"
    >
      Heading
    </span>
     
    <span
      class="some-other-class"
      id="subheader"
      innertext="Sub Heading"
    >
      Sub Heading
    </span>
  </nuxt-link-stub>
</div>
`;

exports[`AppTitleCard Component Tests base component state, matches snapshot 1`] = `
<div
  class="v-toolbar__title"
>
  <nuxt-link-stub
    to="/"
  >
    <span
      class=""
      id="header"
      innertext=""
    >
      
    </span>
     
    <span
      class=""
      id="subheader"
      innertext=""
    >
      
    </span>
  </nuxt-link-stub>
</div>
`;
